src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * This function simulates a chat bot who interacts with a user to keep track of a todo list.
+ *
+ * @author Megan Wee Rui En
+ * @version CS2103T AY21/22 Semester 1
+ */
+package duke;
+
+import duke.command.Command;
+import duke.exceptions.CommandDoesNotExist;
+import duke.exceptions.DukeExceptions;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+import java.util.Scanner;
+import java.io.IOException;
+
+public class Duke {
+
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Ui ui;
+    private final Command command;
+
+    /**
+     * Constructs a Duke object
+     *
+     * @param filePath The location where information will be saved in the project.
+     * @throws IOException
+     */
+    public Duke(String filePath) throws IOException {
+        this.ui = new Ui(); // Performs the self introduction upon successful initialization.
+        this.storage = new Storage(filePath);
+        this.taskList = new TaskList(storage.load());
+        this.command = new Command(storage, ui);
+    }
+
+    /**
+     * This function is the body of the chat bot, and contains the flow and commands.
+     */
+    public void run() throws DukeExceptions, IOException {
+        ui.showWelcome();
+        String userCommand; // this is the container for the full command received from the user
+        String cmd; // this is the container for the first word of the command received from the user
```

Comment:
While I don't think this explicitly violates the coding standard, I think it would be clearer to have a more descriptive name for this variable to differentiate it from `userCommand`, such as `commandWord`.